utils/src/main/scala/za/co/absa/enceladus/utils/broadcast/BroadcastUtils.scala

```diff
@@ -102,338 +97,120 @@ object BroadcastUtils {
+          mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4, p5, p6, p7), defaultValue)
+      }, valueType)
+      case 8 => udf(new UDF8[Any, Any, Any, Any, Any, Any, Any, Any, Any] {
+        override def call(p1: Any, p2: Any, p3: Any, p4: Any, p5: Any, p6: Any, p7: Any, p8: Any): Any =
+          mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4, p5, p6, p7, p8), defaultValue)
+      }, valueType)
+      case 9 => udf(new UDF9[Any, Any, Any, Any, Any, Any, Any, Any, Any, Any] {
+        override def call(p1: Any, p2: Any, p3: Any, p4: Any, p5: Any, p6: Any, p7: Any, p8: Any, p9: Any): Any =
+          mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4, p5, p6, p7, p8, p9), defaultValue)
+      }, valueType)
+      case 10 => udf(new UDF10[Any, Any, Any, Any, Any, Any, Any, Any, Any, Any, Any] {
+        override def call(p1: Any, p2: Any, p3: Any, p4: Any, p5: Any, p6: Any, p7: Any, p8: Any, p9: Any, p10: Any): Any =
+          mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4, p5, p6, p7, p8, p9, p10), defaultValue)
+      }, valueType)
+      case n => throw new IllegalArgumentException(s"Mapping UDFs with $n arguments are not supported. Should be between 1 and 10.")
+    }
+  }
+
+  private def getErrorLambda(mappingTable: Broadcast[LocalMappingTable], outputColumns: Seq[String],
+                             mappings: Seq[Mapping], numberOfParams: Int): UserDefinedFunction = {
+    def applyError(key: Seq[Any]): Any = {
+      if (mappingTable.value.contains(key)) {
         null
       } else {
-        ErrorMessage.confMappingErr(outputColumn, Seq(safeToString(param1), safeToString(param2), safeToString(param3),
-          safeToString(param4), safeToString(param5), safeToString(param6), safeToString(param7), safeToString(param8),
-          safeToString(param9)), mappings)
+        val strings: Seq[String] = key.map(a => safeToString(a))
+        ErrorMessage.confMappingErr(outputColumns.mkString(","), strings, mappings)
       }
     }
-  }
-
-  private def getErrorLambdaParam10(mappingTable: Broadcast[LocalMappingTable],
-                                   outputColumn: String,
-                                   mappings: Seq[Mapping]): AnyRef = {
-    (param1: Any, param2: Any, param3: Any, param4: Any, param5: Any,
-     param6: Any, param7: Any, param8: Any, param9: Any, param10: Any) => {
-      val mt = mappingTable.value.map
-      if (mt.contains(Seq(param1, param2, param3, param4, param5, param6, param7, param8, param9, param10))) {
-        null
-      } else {
-        ErrorMessage.confMappingErr(outputColumn, Seq(safeToString(param1), safeToString(param2), safeToString(param3),
-          safeToString(param4), safeToString(param5), safeToString(param6), safeToString(param7), safeToString(param8),
-          safeToString(param9), safeToString(param10)), mappings)
-      }
+    val errorMessageType = ScalaReflection.schemaFor[ErrorMessage].dataType
```

Comment:
Here  I use Java UDFs, but there is an alternative. Using this alternative, however, doesn't allow reusing a function like `applyError` and only works if it's inline, otherwise will fail with something like "Cannot have return type any". 

The alternative does not need a return type parameter and looks like this:
`case 1 => udf((p1: Any) => if (mappingTable.value.contains(Seq(p1))) {
        null
      } else {
        val strings: Seq[String] = Seq(p1).map(a => safeToString(a))
        ErrorMessage.confMappingErr(outputColumns.mkString(","), strings, mappings)
      })`

@dk1844 , which one would you consider preferable?

Reply:
It seems to me that _doesn't allow reusing a function like `applyError` and only works if it's inline_ is too much of a limitation, so I would probably keep it as is.